packages/x-charts/src/BarChart/checkScaleErrors.ts

```diff
@@ -0,0 +1,54 @@
+import { DEFAULT_X_AXIS_KEY, DEFAULT_Y_AXIS_KEY } from '../constants';
+import { AxisDefaultized, isBandScaleConfig, isPointScaleConfig } from '../models/axis';
+import { SeriesId } from '../models/seriesType/common';
+
+const getXOrY = (isVertical: boolean, isContinuous: boolean, x: string, y: string) => {
+  if (isVertical) {
+    return isContinuous ? y : x;
+  }
+  return isContinuous ? x : y;
+};
+
+const getAxisMessage = (isVertical: boolean, axisKey: string, isContinuous: boolean = false) => {
+  const axisName = getXOrY(isVertical, isContinuous, 'x-axis', 'y-axis');
+  const axisKeyName = getXOrY(isVertical, isContinuous, 'xAxis', 'yAxis');
+  const axisDefaultKey = getXOrY(isVertical, isContinuous, DEFAULT_X_AXIS_KEY, DEFAULT_Y_AXIS_KEY);
+
+  return axisKey === axisDefaultKey
+    ? `The first \`${axisKeyName}\``
+    : `The ${axisName} with id "${axisKey}"`;
+};
```

Comment:
```suggestion
const getAxisMessage = (axisDirection: 'x' | 'y') => {
  const axisName = `${axisDirection}-axis`;
  const axisKeyName = `${axisDirection}Axis`;
  const axisDefaultKey = axisDirection === 'x'? DEFAULT_X_AXIS_KEY : DEFAULT_Y_AXIS_KEY;

  return axisKey === axisDefaultKey
    ? `The first \`${axisKeyName}\``
    : `The ${axisName} with id "${axisKey}"`;
};
```

Reply:
Issue is that we have both `discrete` and `continuous`, and one is the inverse of the other. We also need the proper `axisKey` for the error message.

Reply:
The idea was to move this logic in a single place

```diff
const discreteAxisKey = verticalLayout ? xAxisKey : yAxisKey;
const continuousAxisKey = verticalLayout ? yAxisKey : xAxisKey;
+ const discreteAxisDirection = verticalLayout ? 'x' : 'y';
+ const continuousAxisDirection = verticalLayout ? 'y' : 'x';
```

And then if you do a test on the `discreteAxisKey` you call the function with `discreteAxisDirection`. And if you do a test on `continuousAxisKey` you call the error message with `continuousAxisDirection`

Reply:
🤦